### PR TITLE
🐛 Fixed error caused by accepting invitation with existing email

### LINKED ISSUE #12060: Error when accepting invitation with existing email
### Issue Summary

Accepting a staff invitation email, and submitting an email of an already existing user will cause the DB to throw an error because the email would no longer be unique.

The fix is likely to be to check if the email already exists.

### To Reproduce

1. Invite a new staff user
1. Open the invitation link
1. Use an email of an already existing staff member
1. :boom: Error shown below form

image

```
    HTTP/1.1 500 Internal Server Error
    --
    InternalServerError: insert into `users` (`accessibility`, `bio`, `cover_image`, `created_at`, `created_by`, `email`, `facebook`, `id`, `last_seen`, `locale`, `location`, `meta_description`, `meta_title`, `name`, `password`, `profile_image`, `slug`, `status`, `tour`, `twitter`, `updated_at`, `updated_by`, `visibility`, `website`) values (NULL, NULL, NULL, '2020-07-20 03:24:52', '1', 'secret', NULL, 'id', NULL, NULL, NULL, NULL, NULL, 'my name', 'secret password', NULL, 'secret', 'active', NULL, NULL, '2020-07-20 03:24:52', '1', 'public', NULL) - ER_DUP_ENTRY: Duplicate entry 'edited@email.com' for key 'users_email_unique'
        at new GhostError (/home/ghost/node_modules/@tryghost/errors/lib/errors.js:10:26)
        at _private.prepareError (/home/ghost/core/server/web/shared/middlewares/error-handler.js:53:19)
        at Layer.handle_error (/home/ghost/node_modules/express/lib/router/layer.js:71:5)
        at trim_prefix (/home/ghost/node_modules/express/lib/router/index.js:315:13)
        at /home/ghost/node_modules/express/lib/router/index.js:284:7
        at Function.process_params (/home/ghost/node_modules/express/lib/router/index.js:335:12)
........

    Error: ER_DUP_ENTRY: Duplicate entry 'edited@email.com' for key 'users_email_unique'
        at Query.Sequence._packetToError (/home/ghost/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
        at Query.ErrorPacket (/home/ghost/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
        at Protocol._parsePacket (/home/ghost/node_modules/mysql/lib/protocol/Protocol.js:291:23)
        at Parser._parsePacket (/home/ghost/node_modules/mysql/lib/protocol/Parser.js:433:10)
 ........
```

Originating from Sentry: https://sentry.io/organizations/ghost-foundation/issues/1634959159/events/ea2ce7b6681a4c119104533d78c9520e/

### Technical details:

* Ghost Version: 3.25.0
* Node Version: 12.18.0
* Browser/OS: Firefox/Linux
* Database: SQLite

### PR DESCRIPTION
**Pull request by imox2**
closes #12060
- on invitation, if user uses an email which is already an existing user
  - it causes the DB to throw an error because the email would no longer
    be unique. This was not handled
  - Added an extra check to query user model
    - If user exists, raise a Validation Error
    - else create user

- [x] There's a clear use-case for this code change
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `yarn test` and `yarn lint`)

### CODE REVIEW

File: core/server/services/invitations/accept.js [hunk 1]
```diff
@@ -18,6 +18,15 @@ async function accept(invitation) {
         throw new errors.NotFoundError({message: i18n.t('errors.api.invites.inviteExpired')});
     }
 
+    let user = await models.User.findOne({email: data.email});
+    if (user) {
+        throw new errors.ValidationError({
+            message: i18n.t('errors.api.invites.inviteEmailAlreadyExist.message'),
+            context: i18n.t('errors.api.invites.inviteEmailAlreadyExist.context'),
+            errorDetails: i18n.t('errors.api.invites.inviteEmailAlreadyExist.help')
```
Comment (naz) [inline core/server/services/invitations/accept.js:26]:
```suggestion
            help: i18n.t('errors.api.invites.inviteEmailAlreadyExist.help')
```

The convention is to use `help` property when throwing an error to provide information on how the end user could fix the problem :wink:

**Review by naz — CHANGES_REQUESTED** [review]

File: core/server/services/invitations/accept.js (hunk 1 above)
Comment (imox2) [reply core/server/services/invitations/accept.js:26]: @naz sure , i can make this change. But you are aware that this change will only be visible on the server console and end user would only be shown the string in message (i.e `message: i18n.t('errors.api.invites.inviteEmailAlreadyExist.message')`) . right?

Comment (naz) [reply core/server/services/invitations/accept.js:26]: Yes, displaying a `message` part in the UI is the convention we have right now for most errors like this one. Again, might be changed in the future, but would need to be as a part of a global "error messages revamp". Breaking existing convention for the sake of single case will make it harder to refactor and reason about in the future :)
 
// cc @peterzimon, context above might be useful when reviewing [the copy](https://github.com/TryGhost/Ghost/issues/12060#issuecomment-684178756) for this case

Comment (imox2) [reply core/server/services/invitations/accept.js]: @naz agreed, pushed the changes. Thanks

**Review by peterzimon — APPROVED** [review]

**Comment by imox2** [comment]
@naz did you get a check to see it again? I had requested review from you again. Thanks

**Comment by naz** [comment]
Hey @imox2 :wave: I will check this once again when I free up from other priorities. It's in my queue

**Review by naz — APPROVED** [review]
It's good to :shipit:

**Comment by naz** [comment]
Thanks for well prepared fix @imox2! I'll get around the other PR tomorrow, I hope

**Comment by imox2** [comment]
@naz Thanks for the review and suggestions. This is my first PR ever, Yay!

**Pull request merged**